python/bin/watchman-make

```diff
@@ -16,27 +16,20 @@ def patterns_to_terms(pats):
         terms.append(['match', p, 'wholename', {'includedotfiles': True}])
     return terms
 
-
 class Target(object):
-    """ Represents a makefile target that we'd like to build
+    """ Base Class for a Target
 
     We track the patterns that we consider to be the dependencies for
     this target and establish a subscription for them.
 
     When we receive notifications for that subscription, we know that
-    we should execute the command to build this target.
+    we should execute the command.
     """
-
-    def __init__(self, name, make, targets, patterns):
+    def __init__(self, name, patterns, cmd, triggered=False):
         self.name = name
-        self.make = make
-        self.targets = targets
         self.patterns = patterns
-        self.triggered = False
-
-    def __repr__(self):
-        return '{make=%r targets=%r pat=%r}' % (
-            self.make, self.targets, self.patterns)
+        self.cmd = cmd
+        self.triggered = triggered
```

Comment:
I am assuming you are referring to both `self.make` and `self.targets`. They were only needed to construct the command that was going to run, but now the subclass will set the `self.cmd` which will run. I will set `self.triggered = false` and then remove it from the argument list.